### PR TITLE
RELATED: BB-1569 added itemDescriptionObjects api to catalogue

### PR DESCRIPTION
**Pull request by yanlipnican**
---

# PR checklist

- [x] Change was tested using dev-release in Analytical Designer and Dashboards (if applicable).
- [x] Change is described in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Migration guide (for major update) is written to [CHANGELOG.md](../blob/master/CHANGELOG.md).

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/332/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/398/ : SUCCESS in 1m 44s

File: src/catalogue.ts [hunk 1]
```diff
@@ -52,6 +52,34 @@ const getRequiredDataSets = (options: any) => {
...
+    if (itemDescription.uri) {
+        return itemDescription.uri;
+    }
+    if (itemDescription.expression) {
+        return itemDescription.expression;
+    }
+    throw new Error("Item description can only have expression or uri");
+};
+
+const definitionAndColumnsToItemDescriptionObjects = ({ columns, definitions }: any): any => {
```
Comment (martinmosko) [inline src/catalogue.ts]: hmm what about just `convertToItemDescriptionObjects`, `buildItemDescriptionObjects`

`definitionAndColumnsToItemDescriptionObjects` its too long to me

Comment (martinmosko) [inline src/catalogue.ts]: hmm could we specify return type instead of `any`?

File: test/catalogue.test.ts [hunk 2]
```diff
@@ -319,4 +319,49 @@ describe("Catalogue", () => {
...
+                {
+                    metricDefinition: {
+                        identifier: "/uri/1",
+                        expression: "expression",
+                    },
+                },
+            ],
+        };
+
+        it("it should get itemDescriptionObjects and unwrap them to strings", async () => {
```
Comment (martinmosko) [inline test/catalogue.test.ts]: `it("it`

File: src/catalogue.ts [hunk 3]
```diff
@@ -52,6 +52,34 @@ const getRequiredDataSets = (options: any) => {
     return { requiredDataSets: { type: "PRODUCTION" } };
 };
 
+const unwrapItemDescriptionObject = (itemDescription: any): string => {
```
Comment (martinmosko) [inline src/catalogue.ts]: same here please specify `itemDescription` type

File: src/catalogue.ts [hunk 4]
```diff
@@ -114,6 +142,21 @@ export class CatalogueModule {
         });
     }
 
+    public async loadItemDescriptionObjects(
+        projectId: string,
+        mdObj: any,
+        attributesMap: any,
```
Comment (martinmosko) [inline src/catalogue.ts]: I'm not sure how hard it is but... please try to specify a type for arguments... at least for this new public interface...

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/333/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/399/ : SUCCESS in 1m 18s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/334/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/400/ : SUCCESS in 1m 30s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/336/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/402/ : SUCCESS in 1m 21s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/337/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/403/ : SUCCESS in 1m 08s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-merge-compliance-check/90/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/404/ : SUCCESS in 1m 30s

**Pull request merged**